lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -758,6 +758,27 @@ OpFoldResult BroadcastOp::fold(FoldAdaptor adaptor) {
   return {};
 }
 
+LogicalResult BroadcastOp::verify() {
+  auto src = getSrc();
+  auto srcTensorType = cast<RankedTensorType>(src.getType());
+  auto srcShape = srcTensorType.getShape();
+  auto result = getResult();
+  auto resultTensorType = cast<RankedTensorType>(result.getType());
+  auto resultShape = resultTensorType.getShape();
+  if (srcShape.size() > resultShape.size()) {
```

Comment:
The two shapes should match exactly in size? Otherwise `expand_dim` should be used.

Reply:
I agree with your point about ```expand_dim``` though. With the stricter check, the programmer could achieve the same result by chaining ```expand_dim``` followed by ```broadcast```. The relaxed ```broadcast``` could be trying to save one extra call to ```expand_dim```.

Reply:
I see. Let's fix the test instead. I think it's an invalid IR since it's going to hit the assert during llvm lowering in practice: https://github.com/triton-lang/triton/blob/3b035657139ddfe9cc03f03d051193edb57214d5/lib/Conversion/TritonGPUToLLVM/ViewOpToLLVM.cpp#L326

Reply:
Done.